### PR TITLE
[ONNXImporter] Add support for ONNX Loop operator by unrolling in loader.

### PR DESCRIPTION
**Pull request by aturetsk**
Author: Jun Bum Lim <junlim@cadence.com>

Summary:
Add support for ONNX Loop operator by unrolling in loader. This approach supports only static loops where the max trip-count and condition is constant or foldable to constant during model loading time.
While this might not always be the best solution from performance perspective, it allows to run new types of networks in Glow.

Documentation:
N/A

Test Plan:
Added ONNX importer unit tests.

### CODE REVIEW

**Comment by mciprian13** [comment]
@aturetsk This sound awesome. Do you know if with this update whether the model Yolo v3 tiny works with Glow?
https://github.com/onnx/models/tree/master/vision/object_detection_segmentation/tiny-yolov3
Thanks!

**Comment by aturetsk** [comment]
@mciprian13, I haven't tried to run TinyYolo model.

**Comment by mciprian13** [comment]
@aturetsk I assume this operator enables some new models with Glow. Can you give some examples? Thanks!

**Comment by aturetsk** [comment]
@mciprian13, I'm not aware of any particular model which would start working with Glow after this MR. I've seen loops in some complex models, like FRCNN and MTCNN. Unrolling is not always possible or optimal solution in such models, but having a limited support for Loop operator can help bring up new models.

**Comment by aturetsk** [comment]
Ping

File: lib/Importer/ProtobufLoader.cpp [hunk 1]
```diff
@@ -197,11 +197,17 @@ Error ProtobufLoader::createAndRegisterConstant(llvm::StringRef name,
 
 void ProtobufLoader::deleteUnusedConstants() {
   std::vector<std::string> nodeValuesToRemove;
+  // Note that it's possible a constant is referred by more than one names
+  // (e.g., via Identity operator). Therefore, we mantain a set of constants to
```
Comment (jfix71) [inline lib/Importer/ProtobufLoader.cpp:201]: typo: `mantain -> maintain`

File: lib/Importer/ProtobufLoader.cpp [hunk 2]
```diff
@@ -211,9 +217,12 @@ void ProtobufLoader::deleteUnusedConstants() {
     auto *c = llvm::dyn_cast<Constant>(it->second.getNode());
     DCHECK(c) << "NodeValue with name " << name
               << " was expected to have been a Constant";
```
Comment (jfix71) [inline lib/Importer/ProtobufLoader.cpp:219]:
```suggestion
    DCHECK(llvm::isa<Constant>(it->second.getNode())) << 
              << "NodeValue with name " << name
              << " was expected to have been a Constant";
```

**Review by jfix71 — APPROVED** [review]
Very cool, awesome contribution, thanks for upstreaming it! Looks very clean and well commented, appreciate that. Just added a couple nits.

**Comment by aturetsk** [comment]
@jfix71, thanks for review, all remarks addressed.

**Review by jfix71 — APPROVED** [review]

**Pull request closed without merging**